docs/site/Express-middleware.md

```diff
@@ -355,6 +355,22 @@ router.get('/hello', handler);
 const binding = server.expressMiddleware('middleware.express.greeting', router);
 ```
 
+## Access MiddlewareContext/RequestContext in an Express middleware
+
+In some cases, your Express middleware wants to access LoopBack's RequestContext
+to resolve certain bindings. This can be done via `MIDDLEWARE_CONTEXT` property
+of the Express request object, which is set up by LoopBack when the
+RequestContext is instantiated.
+
+```ts
+import {MIDDLEWARE_CONTEXT, RequestContext} from '@loopback/rest';
+
+function expressHandler(req, res, next) {
+  const reqCtx = (req as any)[MIDDLEWARE_CONTEXT];
```

Comment:
It makes me wonder if we can further improve this solution to make the access more ergonomic? How about something like this - notice we don't have to cast `req` to `any`:

```ts
import {getExpressRequestContext} from '@loopback/rest';

function expressHandler(req, res, next) {
  const ctx = getExpressRequestContext(req);
  // ...
}
```

Personally, I am not very familiar with the new middleware-based architecture of our REST layer. Is there any difference between middleware context and per-request context? So far, I was used to per-request context, that's why I prefer something like `getExpressRequestContext` so that I don't have to understand a new concept of middleware context. Maybe it's just me and new users will be already familiar with middleware context, in which case we could use a name like `getMiddlewareRequestContext`?

Reply:
RequestContext extends MiddlewareContext. It's fine to have a utility such as `getMiddlewareContext<T extends MiddlewareContext>(req: Request): T`.